### PR TITLE
TurboAPI

### PR DESCRIPTION
**Pull request by Tekinical**
Ok 4th attempt, I fixed all lint issues, just prettier wasn't working on the GitHub check so I did it myself!<br>TurboAPI can do POST and GET requests from code blocks.<br>
Screenshot 2023-09-04 000447
real trap sh
New Project (5)
<br>If you wanna try it out now, import it via: http://createserver.xyz/turbowarp/TurboAPI.js

### CODE REVIEW

**Comment by Tekinical** [comment]
@LilyMakesThings my bad, this is my first time attempting pull requests, had no idea you could re-edit them, next time I will.

**Comment by GarboMuffin** [comment]
don't close and open a new pull request, just edit the existing pull request, otherwise you are sending us spam

**Review by NexusKitten — CHANGES_REQUESTED** [review]
Is there any reason to use this over [Network](https://turbowarp.org/editor?extension=https://extensions.turbowarp.org/true-fantom/network.js)? According to CONTRIBUTING.md, extensions that are similar to pre-existing ones are less likely to get accepted, and both blocks in this extension have replicated functionality over there:

image

This doesn't mean take down the PR, but it may not be approved due to their similarities.

File: extensions/turboapi.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+                defaultValue: "Arguments",
+              },
+            },
+          },
+          {
+            opcode: "responseAPI",
+            blockType: Scratch.BlockType.REPORTER,
+            text: "API Response",
+            blockIconURI:
+              "https://freeiconshop.com/wp-content/uploads/edd/download-flat.png",
```
[the remark this answers is not in the document]
Comment (Tekinical) [reply extensions/turboapi.js:51]: I'll work on all these changes soon, working on something atm.

File: extensions/turboapi.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+  let APIResponse = null;
+  let JsonThing = null;
+
+  class TurboAPI {
+    getInfo() {
+      return {
+        id: "TurboAPI",
+        name: "TurboAPI",
+        menuIconURI: "https://files.catbox.moe/4eb196.png",
```
[the remark this answers is not in the document]
Comment (Tekinical) [reply extensions/turboapi.js:17]: ✅

File: extensions/turboapi.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Tekinical) [reply extensions/turboapi.js:51]: ✅

**Comment by Tekinical** [comment]
@NexusKitten every suggestion you made is now resolved, I'll run it through prettifier then publish it.

**Comment by Tekinical** [comment]
All changes done, updated the extension (also updated the 3rd party link to the newest version too)

**Comment by mightbelite** [comment]
I didn't see this comment, sorry. This was my first time making a pull request so I had no idea it could auto-update.

**Comment by mightbelite** [comment]
> I didn't see this comment, sorry. This was my first time making a pull request so I had no idea it could auto-update.

(This is tekinical on his alt)

**Comment by Tekinical** [comment]
I'm in the process of updating it to be more lightweight, but for some reason, this doesn't report a value?
image
image
image

**Comment by CST1229** [comment]
> I'm in the process of updating it to be more lightweight, but for some reason, this doesn't report a value?

It probably throws an error, and [unsandboxed blocks shouldn't throw.](https://docs.turbowarp.org/development/extensions/unsandboxed#increased-power-brings-increased-responsibility)

**Comment by Tekinical** [comment]
> > I'm in the process of updating it to be more lightweight, but for some reason, this doesn't report a value?
> 
> It probably throws an error, and [unsandboxed blocks shouldn't throw.](https://docs.turbowarp.org/development/extensions/unsandboxed#increased-power-brings-increased-responsibility)

Already checked and it doesn't. It just simply doesn't report the value.

**Comment by Tekinical** [comment]
I just wont add this then; I'll figure out something else to make the extension better.

**Comment by NexusKitten** [comment]
> I'm in the process of updating it to be more lightweight, but for some reason, this doesn't report a value? image image image

You forgot to make the function async

**Comment by Tekinical** [comment]
> > I'm in the process of updating it to be more lightweight, but for some reason, this doesn't report a value? image image image
> 
> You forgot to make the function async

Thank you, update to the script that makes it more lightweight will be published in a minute.
image
image

**Comment by Tekinical** [comment]
Passed all checks! Yippee!

(if anyone wants to try this out, you can copy it here or from my fork. http://createserver.xyz/turbowarp/TurboAPI.js

**Comment by GarboMuffin** [comment]
please submit exactly 1 extension per pull request

**Comment by Tekinical** [comment]
> please submit exactly 1 extension per pull request

how do i do that. it automatically put my extension in this pull request

**Comment by Tekinical** [comment]
! Before commenting on this: I am no longer giving support to this extension unless it gets added to the extension gallery, this was easy to make and I don't wanna add anything else. It's basically fetch but with a post request option. Go check out my Save Codes extension which is much better than this.

**Pull request closed without merging**